theano/sandbox/cuda/opt.py

```diff
@@ -446,6 +447,26 @@ def local_gpu_dot_to_dot22(node):
     return False
 
 
+@local_optimizer(None)
+def local_assert_no_cpu_op(node):
+    if not isinstance(node.op, GpuOp) and all([var.owner and isinstance(var.owner.op,
+        HostFromGpu) for var in node.inputs]) and all([var.owner and
+            isinstance(var.owner.op, GpuFromHost) for var in node.outputs]):
```

Comment:
So, here, `var.owner.op for var in node.outputs` will be `node.op`, so not `GpuFromHost`.
You would have to check in `node.outputs.clients` to see what the next operation is, and if it is a `GpuFromHost`.
I'm not sure what to do if an output has more than one client, and some of them are `GpuFromHost` and some are not. @nouiz?